### ISSUE TITLE
The PLACE_TYPE-create and PLACE_TYPE-edit xlsx files need attention with flexible hierarchy. 

### ISSUE THREAD

**Issue by newtewt** [medic/medic#5884]
This is more of a discussion and possible spinning out more issues as there are a few things affected by configurable hierarchy changes and how medic-conf works. 

The forms in `/config/standard/forms/contact/PLACE_TYPE-create.xlsx` and `/config/standard/forms/contact/PLACE_TYPE-edit.xlsx` only work with the default config hierarchy. Should these be updated to be more flexible to work with custom contact_types in configurable hierarchy, removed, or nothing and whomever makes these custom types should supply their own default form?

Secondly, medic-conf seems to overwrite the files in `/config/standard/forms/contact` when those files and the `place-types.json` file exists. This seems like a pretty destructive change and very confusing if you are copying in or making changes to an xlsx locally. I would hope we could warn or possibly add an action/flag to specifically overwrite. Also, we should probably document this.

**Comment by garethbowen** [comment]
> The forms in /config/standard/forms/contact/PLACE_TYPE-create.xlsx and /config/standard/forms/contact/PLACE_TYPE-edit.xlsx only work with the default config hierarchy. Should these be updated to be more flexible to work with custom contact_types in configurable hierarchy, removed, or nothing and whomever makes these custom types should supply their own default form?

The standard config ships with the default hierarchy configured implicitly therefore I think it's ok to ship forms that work with that config. If someone wants a different hierarchy they can provide their own forms. Presumably the CHT reference app will be different again...

> Secondly, medic-conf seems to overwrite the files in /config/standard/forms/contact when those files and the place-types.json file exists. This seems like a pretty destructive change and very confusing if you are copying in or making changes to an xlsx locally. I would hope we could warn or possibly add an action/flag to specifically overwrite. Also, we should probably document this.

Yeah that's strange. I think we should just do nothing if the forms exist because you probably never want to go back to the original. It's much more likely that you started there and made custom changes which would be destroyed next time this runs.

I'm actually wondering if we should get rid of the `PLACE_TYPE-create.xslx` and `-edit.xlsx`. As far as I can tell it creates the same forms for all place types (with minimal changes) which is almost certainly not useful for anyone. We should ship example forms (as we do) which can be manually copied and edited if desired.

@kennsippell What do you think?

**Comment by kennsippell** [comment]
imo - now is a good time to make breaking changes since v3.0 isn't yet announced broadly. We can batch this removal in with that breaking change announcement.

I spoke with @binokaryg because the CHT-Ref-App currently uses the default hierarchy and uses a `PLACE_TYPE-edit.xlsx`. There is a short-term plan to change that hierarchy and remove the file, so no blocker there.

**Comment by garethbowen** [comment]
> the CHT-Ref-App currently uses the default hierarchy and uses a PLACE_TYPE-edit.xlsx

Are the generated place creation and editing forms also committed? If so, rerunning this code would replace those forms with the autogenerated ones again.

**Comment by kennsippell** [comment]
https://github.com/medic/medic-projects/tree/cht-app/cht/forms/contact

**Comment by garethbowen** [comment]
In my opinion no configuration including the reference app should include both the `place-types` and all the generated forms as it's just asking for custom changes to be overwritten.

I propose in the short term I change the form generation code to only generate if the forms don't already exist. I think that's a bug fix not a breaking change because it's more aligned with how I expect it to behave.

@newtewt Does that make sense to you?

**Comment by kennsippell** [comment]
> In my opinion no configuration including the reference app should include both the place-types and all the generated forms as it's just asking for custom changes to be overwritten.

Agree. FYI to @binokaryg and I will add this to the code review feedback for CHT-ref-app.

> only generate if the forms don't already exist

That would solve problem 2 listed above. But I think you'd have to also extend it to generate forms based on the configurable hierarchy to solve both issues.

**Comment by garethbowen** [comment]
> That would solve problem 2 listed above. But I think you'd have to also extend it to generate forms based on the configurable hierarchy to solve both issues.

I don't think so. If you remember to edit the place-types.json then you can generate forms for any hierarchy.

**Comment by garethbowen** [comment]
I accidentally merged to master before AT (sorry!). Please AT again `master`.
